### ISSUE TITLE
[BUG-REPORT] Azure Data Lake Storage Gen2 using fsspec

### ISSUE THREAD

**Issue by vernetya** [vaexio/vaex#1272]
Hi,

I'm trying to work with Vaex on Azure Data Lake Storage Gen2. Thanks to recent fix (#1248) I managed to have it somehow work but in best case the performance are very poor. Doing the same exercise works very well on GCS.

In my case, I have a jupyterlab deployed on a kubernetes cluster in Azure, in the same region, in Azure storage I have a single hdf5 file of about ~2Go. Then I simply try to do an `vaex.open` and I'm using [adlfs](https://github.com/dask/adlfs). Here's a code snippet I'm running:

```
import vaex
import adlfs

storage_account = "..."
account_key = "..."
container = "..."
object_path = "..."

fs = adlfs.AzureBlobFileSystem(account_name=storage_account, account_key=account_key)
df = vaex.open(f"abfs://{container}/{object_path}")
```

It successfully opens it but it takes about 3 minutes. Simply downloading the entire file takes less than a minute using the very same file system like so:

```
fs.download(f"abfs://{container}/{object_path}", local_path)
```

If I do the same on Google Cloud, similar setup, jupyterlab deployed in GKE cluster and same stored in GCS same region `vaex.open` takes less than 2 seconds.

I have the same behavior and slowness with parquet file instead of hdf5 on Azure.

I don't know if I'm doing something wrong here that could explain such difference in performance.

**Software information**
versions:
adlfs                            0.6.3
pyarrow                       3.0.0
pandas                        1.2.3
numpy                         1.20.1
tables                        3.6.1
vaex                          4.1.0
vaex-astro                    0.8.0
vaex-core                     4.1.0
vaex-hdf5                     0.7.0
vaex-jupyter                  0.6.0
vaex-ml                       0.11.0
vaex-server                   0.4.0
vaex-viz                      0.5.0

**Comment by maartenbreddels** [comment]
Yes, I understand that it can be slow because it is not using the caching system.

Maybe we can build this in .. that would then be the only optional setting we could support when we support other filesystems.

For now, we can do it similar as in https://github.com/vaexio/vaex/blob/a1349ed40afe38556f3e08a20ca1139b73c4c141/packages/vaex-core/vaex/file/s3fs.py#L87


```python
import adlfs
import vaex
import pyarrow as pa
import pyarrow.fs
from vaex.file.cache import FileSystemHandlerCached

fs = adlfs.AzureBlobFileSystem(...)
fs = pa.fs.FSSpecHandler(fs)
fs = FileSystemHandlerCached(fs, scheme='abfs', for_arrow=True)
# fs = pa.fs.PyFileSystem(fs)  # we don't need this anymore 
container = 'foo'
name = 'bar'
df = vaex.open(f'{container}/{name}', fs=fs)
```

Maybe you can take a look how this is done with s3fs, and make support for abfs and open a PR?

Regards,

Maarten
